src/decoders.rs

```diff
@@ -0,0 +1,55 @@
+use crate::levenshtein_functions::{get_delta_action, Action};
+
+pub trait Decoder {
+    fn decode_chunk(&self, parent_data: Vec<u8>, delta_code: &[u8]) -> Vec<u8>;
+}
+
+pub struct LevenshteinDecoder;
+
+impl Decoder for LevenshteinDecoder {
+    fn decode_chunk(&self, mut parent_data: Vec<u8>, delta_code: &[u8]) -> Vec<u8> {
+        let mut buf = [0u8; 4];
+        let mut byte_index = 4;
+
+        while byte_index < delta_code.len() {
+            buf.copy_from_slice(&delta_code[byte_index..byte_index + 4]);
+            let delta_action = u32::from_be_bytes(buf);
+
+            let (action, index, byte_value) = get_delta_action(delta_action);
+            match action {
+                Action::Del => {
+                    parent_data.remove(index);
+                }
+                Action::Add => parent_data.insert(index, byte_value),
+                Action::Rep => parent_data[index] = byte_value,
+            }
+            byte_index += 4;
+        }
+        parent_data
+    }
+}
+
+pub struct GdeltaDecoder;
+impl Decoder for GdeltaDecoder {
```

Comment:
```suggestion
pub struct GdeltaDecoder;

impl Decoder for GdeltaDecoder {
```

и в методе ниже бы пробельчиков добавить для читаемости